From b514cf7ae7b97cb0f8b1f00210df889c06084f05 Mon Sep 17 00:00:00 2001
From: Nithya Balachandran <nbalacha@redhat.com>
Date: Sun, 25 Oct 2015 12:10:35 +0530
Subject: [PATCH 45/67] cluster/dht : op_ret not set correctly in dht_fsync_cbk

local->op_ret was not set correctly in dht_fsync_cbk in case
the file was being migrated

upstream patch : http://review.gluster.org/12401

Change-Id: I209bf71f845526226490d24def04c21bf32390bb
BUG: 1274595
Signed-off-by: Nithya Balachandran <nbalacha@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/60106
Reviewed-by: Dan Lambright <dlambrig@redhat.com>
Tested-by: Dan Lambright <dlambrig@redhat.com>
---
 xlators/cluster/dht/src/dht-inode-read.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xlators/cluster/dht/src/dht-inode-read.c b/xlators/cluster/dht/src/dht-inode-read.c
index 3480151..b832c95 100644
--- a/xlators/cluster/dht/src/dht-inode-read.c
+++ b/xlators/cluster/dht/src/dht-inode-read.c
@@ -807,7 +807,7 @@ dht_fsync_cbk (call_frame_t *frame, void *cookie, xlator_t *this, int op_ret,
                 goto out;
         }
 
-        local->op_errno = op_errno;
+        local->op_ret = op_ret;
         inode = local->fd->inode;
 
         dht_inode_ctx_get_mig_info (this, inode, &src_subvol, &dst_subvol);
-- 
1.7.1

